PURL-TYPES.rst

```diff
@@ -498,6 +498,18 @@ including container images built by Docker and others:
       pkg:oci/static@sha256%3A244fd47e07d10?repository_url=gcr.io/distroless/static&tag=latest
       pkg:oci/hello-wasm@sha256%3A244fd47e07d10?tag=v1
 
+opam
+----
+``opam`` for OCaml packages
+- The default repository is ``https://opam.ocaml.org/``.
+- The ``name`` is the package name. It is not case sensitive and must be lowercased.
```

Comment:
package names are case sensitive and can use upper-case characters.
See the manual:
> Package names have the same restrictions as idents

https://opam.ocaml.org/doc/Manual.html#Package-Formulas

TL;DR:
```
<ident>         ::= { <identchar> }* <letter> { <identchar> }*
<identchar>     ::= <letter> | <digit>  | "_" | "-" | "+"
<letter>        ::= "a".."z" | "A".."Z"
<digit>         ::= "0".."9"
```

Reply:
@kit-ty-kate fixed!